Week1/assets/index.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+{
+
+
+  function fetchJSON(url) {
+    return new Promise(function(resolve, reject) {
+      var xhr = new XMLHttpRequest();
+     
+      xhr.open('GET', url);
+      xhr.responseType = 'json';
+
+      xhr.onload = () => {
+        if (xhr.status < 400) {
+          resolve(xhr.response);
+        } else {
+          reject(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
+        }
+      };
+      xhr.onerror = () => reject(new Error('Network request failed'));
+      xhr.send();
+    });
+  }
+
+
+  function createAndAppend(name, parent, options = {}) {
+    const elem = document.createElement(name);
+    parent.appendChild(elem);
+    Object.keys(options).forEach((key) => {
+      const value = options[key];
+      if (key === 'html') {
+        elem.innerHTML = value;
+      } else {
+        elem.setAttribute(key, value);
+      }
+    });
+    return elem;
+  }
+  const repoLabels = ['Repository', 'Description', 'Forks', 'Updated'];
+  const repoValues = ['name', 'description', 'forks', 'updated_at'];
+  function renderRepo(repo)
+  {
+    const content = document.getElementById('content');
+    content.innerHTML = "";
+    const repoContent = createAndAppend('div', content, {class: 'item', id: 'repoInfo'});
+    const contributorsContent = createAndAppend('div', content, {class: 'item', id: 'contribInfo'});
+    createAndAppend('h2', contributorsContent, {html: 'Constributions'})
+
+    
+    repoLabels.forEach((element, index) => {
+
+      const item = createAndAppend('div', repoContent, {class:'item'});
+      createAndAppend('h3', item, {html: element +': '});
+      
+      const item1 = createAndAppend('div', repoContent, {class:'item'});
+     
+      if(element === 'Repository')
+        createAndAppend('a', item1, {html: repo[repoValues[index]], href: repo.url, target: '_blank'});
+      else if(element === 'Updated')
+      {
+        let myDate = new Date(repo[repoValues[index]]);
+        createAndAppend('h4', item1, {html: myDate.toLocaleString()});
+      }
+      else
+        createAndAppend('h4', item1, {html: repo[repoValues[index]]});
+    });
+  }
+
+  function renderContrib(repo)
+  {
+    const contributors = document.getElementById('contribInfo');
+    fetchJSON(repo.contributors_url).then(function(response){
+      response.forEach((contributor, index)=>{
+        if(index >= 10) return;
```

Comment:
there is no really need to limit the amount of the rendered data on the screen unless you have a hundreds latterly of them.